### PR TITLE
Add "Pass parameters to tests"

### PR DESCRIPTION
**Pull request by DIRECTcut**
#2 "Pass arguments to tests"

### CODE REVIEW

File: examples/pass-parameters-to-tests/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { ClientFunction } from 'testcafe';
+
+const getDocumentUri = ClientFunction(() => document.documentURI);
+
+fixture `Pass parameters to tests`
+
+process.env.Url = './src/env.html';
+
+test('Get URL from an environment variable', async t => {
+    console.log(`Test page URL: ${process.env.Url}`);
```
Comment (miherlosev) [inline examples/pass-parameters-to-tests/index.js:12]: Remove the `console.log` statement and use the strict URL comparison for all tests.

```js
import { pathToFileURL } from 'url';
import { resolve } from 'path';

const getFullFilePath = path => {
    const resolvedPath = resolve(__dirname, path);

    return pathToFileURL(resolvedPath).href;
};
...
test('Navigate to URL from an environment variable', async t => {
      await t.expect(getDocumentUri()).eql(getFullFilePath('./src/env.html'));
});

```

File: examples/pass-parameters-to-tests/src/env.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <span id="span">This is a test page</span>
```
Comment (miherlosev) [inline examples/pass-parameters-to-tests/src/env.html:9]: Add an unique text for each page.

```js
<h1>The page URL loaded from environment variable</h1>
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/pass-parameters-to-tests/index.js (hunk 1 above)
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/index.js:12]: Isnt that more logical:

`test('Navigate to URL from an environment variable', async t => {
      await t.expect(getDocumentUri()).eql(getFullFilePath(**process.env.url**));
});`

File: examples/pass-parameters-to-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+This example demonstrates how to pass parameters to TestCafe tests from the following sources:
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before the tests, the `.json` and `.js` configuration files are imported with JavaScript [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) syntax. The environment variable is declared with the [process.env](https://nodejs.org/api/process.html#process_process_env) property. These values are then available inside tests.
+
+During each of the tests, the `test.page` method reads one of these values to decide the page where the test starts. A `clientFunction` that is declared before the tests begin obtains the full URL of the current page. The `t.expect.eql` method compares this value to the full path to the file that the configuration file or the environment variable specified.
```
Comment (miherlosev) [inline examples/pass-parameters-to-tests/README.md:13]: For each test, the initial page is set by calling to the [test.page](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/page.html) method with appropriate parameter.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/pass-parameters-to-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Pass Parameters to Tests
+
+**Test Code**: [index.js](index.js)
+
+This example demonstrates how to pass parameters to TestCafe tests from the following sources:
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:5]: > This example demonstrates how to pass parameters to TestCafe tests. The parameter values can be obtained from the following sources:

File: examples/pass-parameters-to-tests/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before the tests, the `.json` and `.js` configuration files are imported with JavaScript [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) syntax. The environment variable is declared with the [process.env](https://nodejs.org/api/process.html#process_process_env) property. These values are then available inside tests.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:11]: > Before test execution, the parameters were obtained: two are read from `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement, the third is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.

File: examples/pass-parameters-to-tests/README.md (hunk 3 above)
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:13]: > For each test, a `test.page` method call with an appropriate parameter sets the initial page. A `clientFunction` that is declared before the tests begin obtains the full URL of the current page. The `t.expect.eql` method compares this value to the full path to the file that the configuration file or the environment variable specified.

File: examples/pass-parameters-to-tests/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+This example demonstrates how to pass parameters to TestCafe tests. The parameter values can be obtained from the following sources:
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before test execution, the parameters are obtained. Two of them are loaded from `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement, the third is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.
+
+For each test, a `test.page` method call with an appropriate parameter sets the initial page. A `clientFunction` obtains the [File URL](https://nodejs.org/api/url.html#url_url_pathtofileurl_path) of the current page. The `t.expect.eql` method compares this value to the full path to the file that the configuration file or the environment variable specified.
+
```
Comment (miherlosev) [inline examples/pass-parameters-to-tests/README.md:14]: The `t.expect.eql` method compares this value to the full path calculated based on parameter.

File: examples/pass-parameters-to-tests/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+const getDocumentUri = ClientFunction(() => document.documentURI);
+const getFullFilePath = path => {
+    const resolvedPath = resolve(__dirname, path);
+
+    return pathToFileURL(resolvedPath).href;
+};
+
+fixture `Pass parameters to tests`
+
+//declare an environment variable
```
Comment (miherlosev) [inline examples/pass-parameters-to-tests/index.js:16]: `//set an environment variable`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/pass-parameters-to-tests/README.md (hunk 6 above)
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:14]: > The `t.expect.eql` method compares this value to the full path calculated based on the specified parameter.

File: examples/pass-parameters-to-tests/index.js (hunk 7 above)
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/index.js]: edited

**Review by miherlosev — APPROVED** [review]

File: examples/pass-parameters-to-tests/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+This example demonstrates how to pass parameters to TestCafe tests. The parameter values can be obtained from the following sources:
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before test execution, the parameters are obtained. Two of them are loaded from `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement, the third is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.
+
+For each test, a `test.page` method call with an appropriate parameter sets the initial page. A `clientFunction` obtains the [File URL](https://nodejs.org/api/url.html#url_url_pathtofileurl_path) of the current page. The `t.expect.eql` method compares this value to the full path calculated based on the specified parameter.
```
Comment (VasilyStrelyaev) [inline examples/pass-parameters-to-tests/README.md:13]: > For each test, a `test.page` method call with an appropriate parameter sets the initial page.

a method call with a parameter sets... that's a bit hard to parse... Maybe we can simplify:

*For each test, the `test.page` method sets the initial page. The page URL is passed as its parameter. Then, a client function obtains the [File URL](https://nodejs.org/api/url.html#url_url_pathtofileurl_path) of the current page, and the `t.expect.eql` method compares this value to the full path calculated based on the specified parameter.*

Joined the last two sentences together to avoid too many short phrases.

**Review by VasilyStrelyaev — CHANGES_REQUESTED** [review]

File: examples/pass-parameters-to-tests/README.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Pass Parameters to Tests
+
+**Test Code**: [index.js](index.js)
+
+This example demonstrates how to pass parameters to TestCafe tests. The parameter values can be obtained from the following sources:
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:5]: > You can obtain the parameter values from the following sources...

File: examples/pass-parameters-to-tests/README.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before test execution, the parameters are obtained. Two of them are loaded from `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement, the third is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:11]: edited

File: examples/pass-parameters-to-tests/README.md (hunk 8 above)
Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:13]: I agree. Except:

> For each test, the test.page method sets the initial page. The page URL is passed as its **parameter**. Then, a client function obtains the File URL of the current page, and the t.expect.eql method compares this value to the full path calculated based on the **specified parameter**.

It sounds like we talk about the same parameter, doesn't it? But they are not the same:

> test.page(parameter1) vs getFullFilePath(parameter2)

Comment (DIRECTcut) [reply examples/pass-parameters-to-tests/README.md:13]: edited

**Review by VasilyStrelyaev — APPROVED** [review]

File: examples/pass-parameters-to-tests/README.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+**Test Code**: [index.js](index.js)
+
+This example demonstrates how to pass parameters to TestCafe tests. You can obtain the parameter values from the following sources:
+
+* a JSON configuration file
+* a JavaScript object
+* an environment variable
+
+Before the test execution, two parameters are loaded from the `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement. The third parameter is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.
```
Comment (dirk-pieterse) [inline examples/pass-parameters-to-tests/README.md:11]:
```suggestion
Before test execution, two parameters are loaded from the `.json` and `.js` files with an [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) statement. The third parameter is set as an environment variable with the [process.env](https://nodejs.org/api/process.html#process_process_env) property.
```

File: examples/pass-parameters-to-tests/README.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+For each test, the `test.page` method sets the initial page. The page URL is passed as its parameter. Then, a `clientFunction` obtains the [File URL](https://nodejs.org/api/url.html#url_url_pathtofileurl_path) of the current page, and the `t.expect.eql` method compares this value to the full path calculated based on the specified parameter.
```
Comment (dirk-pieterse) [inline examples/pass-parameters-to-tests/README.md:13]:
```suggestion
The `test.page` method sets the initial page f or each test. The page URL is passed as its parameter. Then, a `clientFunction` obtains the [File URL](https://nodejs.org/api/url.html#url_url_pathtofileurl_path) of the current page, and the `t.expect.eql` method compares this value to the full path calculated based on the specified parameter.
```

**Review by dirk-pieterse — APPROVED** [review]
